index.js

```diff
@@ -31,402 +39,627 @@ console.error = ui.debugLog;
-        throw "Selected device not connected";
-    }
-    return devices[currentDeviceIndex].device;
-}
 
-function replaceDevice(path, newDevice) {
-    const idx = devices.findIndex(d => d.device.originalDescriptor.path === path)
-    if ((idx === -1) && !newDevice) {
-        // nothing to do, as we're trying to remove a device that never existed
-    } else if ((idx > -1) && !newDevice) {
-        // just remove the existing device
-        devices.splice(idx, 1);
-    } else if (idx === -1) {
-        // new device. Add to list.
-        devices.push(newDevice);
-    } else {
-        // existing device. Replace.
-        devices.splice(idx, 1, newDevice);
-    }
+function onChange (features) {
+  if (features == null) throw "no features on connect"
+  devices[features.device_id] = features
+}
 
-    if (currentDeviceIndex >= devices.length) {
-        currentDeviceIndex = Math.max(devices.length -1, 0);
-        uiActions.changeActiveDevice(currentDeviceIndex);
-    } else if (currentDeviceIndex === idx) {
-        uiActions.changeActiveDevice(currentDeviceIndex);
-    } else if ((idx > -1) && (idx < currentDeviceIndex)) {
-        currentDeviceIndex--;
-        uiActions.changeActiveDevice(currentDeviceIndex);
-    }
+function onDisconnect (id) {
+  delete devices[id]
 }
 
-function loadDeviceList() {
-    devList = new trezor.DeviceList({ debug });
+function noDevice () {
+  if (Object.keys(devices).length == 0) {
+    log("No devices.");
+    return true
+  }
+  return false
+}
 
-    log("Got device list");
-    devList.on("connect", device => {
-        log("Device connected", device.features.device_id);
-        replaceDevice(device.originalDescriptor.path, { state: "connected", device })
-        setDeviceListeners(device);
-        if (devices.length === 1) {
-            ui.setActiveDeviceLabel(sprintf("0 '%s'", device.features.label));
-        }
-    });
-    devList.on("error", err => {
-        if (err instanceof Error) {
-            log("Error:", err.message);
-            debugLog(err.stack);
-        } else {
-            log("Error:", err);
-        }
-    });
-    devList.on("connectUnacquired", device => {
-        log("Detected device in use");
-        replaceDevice(device.originalDescriptor.path, { state: "unacquired", device })
-    });
-    devList.on("disconnectUnacquired", device => {
-        log("Unacquired device disconnected");
-        replaceDevice(device.originalDescriptor.path, null);
-    });
-    devList.on("disconnect", device => {
-        log("Device disconnected", device.features.device_id);
-        replaceDevice(device.originalDescriptor.path, null);
-    });
-    devList.on("transport", t => {
-        log("Transport obtained", t.activeName, t.version);
-    });
+function initTrezor() {
+  session.init({
+    connectSrc: 'https://localhost:8088/',
+    lazyLoad: true,
+    popup: false,
+    manifest: {
+      email: 'joegruffins@gmail.com',
+      appUrl: 'https://github.com/decred/decrediton',
```

Comment:
ughhh... 

Are these obligatory with the new trezor connect?

Reply:
Yes...

Reply:
https://github.com/trezor/connect/blob/develop/docs/index.md
`Starting with Trezor Connect 7, we have implemented a new feature — Trezor Connect Manifest — which requires that you as a Trezor Connect integrator, to share your e-mail and application url with us. This provides us with the ability to reach you in case of any required maintenance. This subscription is mandatory.`